test/ui/CategoryFacet/CategoryFacetSearchTest.ts

```diff
@@ -0,0 +1,191 @@
+      categoryFacetSearch = new CategoryFacetSearch(categoryFacetMock);
+    });
+
+    afterEach(() => {
+      _.debounce = realDebounce;
+    });
+
+    it('when building returns a container', () => {
+      const container = categoryFacetSearch.build();
+      expect(container.hasClass('coveo-category-facet-search-container')).toBe(true);
+    });
+
+    it('focus moves the focus to the input element', () => {
+      categoryFacetSearch.facetSearchElement = { input: jasmine.createSpyObj('input', ['focus']) } as any;
+      spyOn(categoryFacetSearch.facetSearchElement, 'input');
+
+      categoryFacetSearch.focus();
+
+      expect(categoryFacetSearch.facetSearchElement.input.focus).toHaveBeenCalled();
+    });
+
+    it('renders values on focus', done => {
+      categoryFacetSearch.build();
+      categoryFacetSearch.focus();
+      setTimeout(() => {
+        expect(categoryFacetSearch.facetSearchElement.searchResults.innerHTML).not.toEqual('');
+        done();
+      });
+    });
+
+    it('renders values correctly', done => {
+      categoryFacetSearch.build();
+
+      categoryFacetSearch.displayNewValues();
+
+      setTimeout(() => {
+        const searchResults = categoryFacetSearch.facetSearchElement.searchResults;
+        const valueCaptions = $$(searchResults).findAll('.coveo-category-facet-search-value-caption');
+        const pathValues = $$(searchResults).findAll('.coveo-category-facet-search-path');
+        const valueCounts = $$(searchResults).findAll('.coveo-category-facet-search-value-number');
+        for (const i of _.range(fakeGroupByValues.length)) {
+          expect($$(valueCaptions[i]).text()).toEqual(`value${i}`);
+          expect($$(pathValues[i]).text()).toEqual('');
+          expect($$(valueCounts[i]).text()).toEqual((i + 1).toString());
+        }
+        done();
+      });
+    });
+
+    it('renders the path correctly', done => {
+      fakeGroupByValues = FakeResults.createFakeGroupByResult('@field', 'a|b|c', 10).values;
+      categoryFacetMock.categoryFacetQueryController.searchFacetValues = () => new Promise(resolve => resolve(fakeGroupByValues));
+      categoryFacetSearch.build();
+
+      categoryFacetSearch.displayNewValues();
+
+      setTimeout(() => {
+        const pathCaption = $$(categoryFacetSearch.facetSearchElement.searchResults).find('.coveo-category-facet-search-path');
+        expect($$(pathCaption).text()).toEqual('a/b/c0');
+        done();
+      });
+    });
+
+    it('sets the correct classes when there is no results', done => {
+      categoryFacetMock.categoryFacetQueryController.searchFacetValues = () => new Promise(resolve => resolve([]));
+      categoryFacetSearch.build();
+
+      categoryFacetSearch.displayNewValues();
+
+      setTimeout(() => {
+        expect($$(categoryFacetMock.element).hasClass('coveo-no-results')).toBe(true);
+        expect($$(categoryFacetSearch.facetSearchElement.search).hasClass('coveo-facet-search-no-results')).toBe(true);
```

Comment:
Utility function to test for "hasNoResults()" or "hasResults()" etc.

In general, there's a ton of repetitive check throughout all your tests that could be much simpler to read and easier to modify if we need to in the future (eg: If the "coveo-no-results" gets changed to something else)